common/src/main/java/org/neo4j/cypher/export/DatabaseSubGraph.java

```diff
@@ -142,24 +142,39 @@ public CountOptimisedDatabaseSubGraph(Transaction transaction, KernelTransaction
 
     @Override
     public long countsForNode(Label label) {
-        return read.countsForNode(tokenRead.nodeLabel(label.name()));
+        int nodeLabelID = tokenRead.nodeLabel(label.name());
+        if (nodeLabelID < 0) {
+            return 0;
+        }
+        return read.countsForNode(nodeLabelID);
     }
 
     @Override
     public long countsForRelationship(RelationshipType type, Label end) {
-        return read.countsForRelationship(
-                TokenConstants.ANY_LABEL, tokenRead.relationshipType(type.name()), tokenRead.nodeLabel(end.name()));
+        int relTypeID = tokenRead.relationshipType(type.name());
+        int nodeLabelID = tokenRead.nodeLabel(end.name());
+        if (nodeLabelID < 0 || relTypeID < 0) {
+            return 0;
+        }
+        return read.countsForRelationship(TokenConstants.ANY_LABEL, relTypeID, nodeLabelID);
     }
 
     @Override
     public long countsForRelationship(Label start, RelationshipType type) {
-        return read.countsForRelationship(
-                tokenRead.nodeLabel(start.name()), tokenRead.relationshipType(type.name()), TokenConstants.ANY_LABEL);
+        int relTypeID = tokenRead.relationshipType(type.name());
+        int nodeLabelID = tokenRead.nodeLabel(start.name());
+        if (nodeLabelID < 0 || relTypeID < 0) {
+            return 0;
+        }
+        return read.countsForRelationship(tokenRead.nodeLabel(start.name()), relTypeID, TokenConstants.ANY_LABEL);
```

Comment:
Just for symmetry with the previous method:

```suggestion
        return read.countsForRelationship(nodeLabelID, relTypeID, TokenConstants.ANY_LABEL);
```